src/js/components/TabForm.js

```diff
@@ -0,0 +1,200 @@
+import classNames from 'classnames';
+import {Form, Tooltip} from 'reactjs-components';
+import GeminiScrollbar from 'react-gemini-scrollbar';
+import React from 'react';
+
+import GeminiUtil from '../utils/GeminiUtil';
+import SideTabs from './SideTabs';
+
+const METHODS_TO_BIND = [
+  'getTriggerSubmit', 'validateForm', 'handleTabClick', 'handleExternalSubmit'
+];
+
+class TabForm extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {currentTab: '', renderGemini: false};
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+    this.triggerSubmit = function () {};
+  }
+
+  componentWillMount() {
+    this.model = {};
+    this.submitMap = {};
+
+    this.setState({
+      currentTab: Object.keys(this.props.definition)[0]
+    });
+
+    this.props.getTriggerSubmit(this.handleExternalSubmit);
+  }
+
+  componentDidMount() {
+    this.setState({renderGemini: true});
+  }
+
+  componentDidUpdate() {
+    // Timeout necessary due to modal content height updates on did mount
+    setTimeout(() => {
+      GeminiUtil.updateWithRef(this.refs.geminiForms);
+    });
+  }
+
+  handleTabClick(currentTab) {
+    this.setState({currentTab});
+  }
+
+  handleFormSubmit(formKey, formModel) {
+    this.model[formKey] = formModel;
+  }
+
+  handleExternalSubmit() {
+    return this.validateForm();
+  }
+
+  validateForm() {
+    this.buildModel();
+    this.props.onSubmit(this.model);
+    return this.model;
+  }
+
+  buildModel() {
+    Object.keys(this.props.definition).forEach((formKey) => {
+      this.submitMap[formKey]();
+    });
+  }
+
+  getTriggerSubmit(formKey, triggerSubmit) {
+    this.submitMap[formKey] = triggerSubmit;
+  }
```

Comment:
I know this comes from somewhere else, but I really don't like a `get*` method that behaves like a `set` one.